onnxruntime/test/perftest/performance_runner.h

```diff
@@ -37,15 +37,29 @@ struct PerformanceResult {
   std::string model_name;
 
   void DumpToFile(const std::basic_string<ORTCHAR_T>& path, bool f_include_statistics = false) const {
+    bool have_file = !path.empty();
     std::ofstream outfile;
-    outfile.open(path, std::ofstream::out | std::ofstream::app);
-    if (!outfile.good()) {
-      printf("failed to open result file");
-      return;
+
+    if (have_file) {
+      outfile.open(path, std::ofstream::out | std::ofstream::app);
+      if (!outfile.good()) {
+        // at least provide some info on the run
+        std::cerr << "failed to open result file '" << path.c_str() << "'. will dump stats to output.";
+        have_file = false;
+        f_include_statistics = true;
+      }
     }
 
-    for (size_t runs = 0; runs < time_costs.size(); runs++) {
-      outfile << model_name << "," << time_costs[runs] << "," << peak_workingset_size << "," << average_CPU_usage << "," << runs << std::endl;
+    if (have_file) {
+      for (size_t runs = 0; runs < time_costs.size(); runs++) {
+        outfile << model_name << "," << time_costs[runs] << "," << peak_workingset_size << ","
+                << average_CPU_usage << "," << runs << std::endl;
```

Comment:
consider replacing endl with "\n"

Reply:
Replaced most. Left one at the end of each section to ensure things are flushed.

---
In reply to: [434039810](https://github.com/microsoft/onnxruntime/pull/4109#discussion_r434039810) [](ancestors = 434039810)